packages/text-field/src/vaadin-text-field.js

```diff
@@ -147,38 +146,10 @@ export class TextField extends PatternMixin(InputFieldMixin(ThemableMixin(Elemen
     };
   }
 
-  static get delegateAttrs() {
-    return [...super.delegateAttrs, 'maxlength', 'minlength'];
-  }
-
-  static get constraints() {
-    return [...super.constraints, 'maxlength', 'minlength'];
-  }
-
-  constructor() {
-    super();
-    this._setType('text');
-  }
-
-  /** @protected */
-  get clearElement() {
-    return this.$.clearButton;
-  }
-
   /** @protected */
   ready() {
     super.ready();
 
-    this.addController(
-      new InputController(this, (input) => {
-        this._setInputElement(input);
-        this._setFocusElement(input);
-        this.stateTarget = input;
-        this.ariaTarget = input;
-      }),
-    );
-    this.addController(new LabelledInputController(this.inputElement, this._labelController));
```

Comment:
Decided to keep the tooltip logic in the component rather than moving to a `TextFieldMixin`.
This is done to allow external libraries use the mixin without including `TooltipController`.